### PR TITLE
Launch server by version

### PR DESCRIPTION
**Pull request by rlagha**
Add a helper method which allows to generate new functions per installed DPF server to ease choosing which server to launch

### CODE REVIEW

File: src/ansys/dpf/core/server.py [hunk 1]
```diff
@@ -388,3 +390,46 @@ def get_or_create_server(server: BaseServer) -> Union[BaseServer, None]:
     if server:
         return server
     return _global_server()
+
+
+def available_servers():
+    """Searches all available installed DPF servers in the current machine
+    Binds new functions to the server module which helps to choose the appropritae/suitable version
+
+    after this call, you can do the following:
```
Comment (PipKat) [inline src/ansys/dpf/core/server.py:399]:
```suggestion
    """Searches all available installed DPF servers on the current machine.
    
    This method binds new functions to the server module, which helps to choose the appropriate version.

    After calling this method, you can do the following:
```
There is no list of things you can do? Is this just not completed yet?

File: src/ansys/dpf/core/server.py [hunk 2]
```diff
@@ -388,3 +390,46 @@ def get_or_create_server(server: BaseServer) -> Union[BaseServer, None]:
...
+    >>> #server = dpf.server.start_2024_2_server()
+
+
+    Parameters
+    ----------
+    server: BaseServer, None
+
+    Returns
+    -------
+    server: returns the map of available DPF servers.
```
Comment (PipKat) [inline src/ansys/dpf/core/server.py:417]:
```suggestion
    server: map of available DPF servers.
```
Because this is the "Returns" section, we don't need the "Returns the" part in the description for ``server``.

File: tests/test_server.py [hunk 3]
```diff
@@ -316,6 +317,23 @@ def version(self):
 
     print(MockServer(remote_server).version)
     with pytest.raises(
-        ValueError, match="Error connecting to DPF LegacyGrpcServer with version 1.0"
+            ValueError, match="Error connecting to DPF LegacyGrpcServer with version 1.0"
```
Comment (PipKat) [inline tests/test_server.py:320]: A messages should always have a verb and concluding period. They should also be in plain English. Something like this would be better: "Could not connect to DPF legacy gRPC server with version 1.0."

File: tests/test_server.py [hunk 4]
```diff
@@ -316,6 +317,23 @@ def version(self):
...
     ):
         dpf.core.server_types.check_ansys_grpc_dpf_version(MockServer(remote_server), timeout=2.0)
+
+@pytest.mark.skipif(
+    os.name == "posix",
+    reason="can only work with local DPF server install",
+)
+def test_available_servers():
+    out = {}
+    out = server.available_servers()
```
Comment (cbellot000) [inline tests/test_server.py:330]: @rlagha shouldn't you test if all the values of  `out` can be called?

**Review by cbellot000 — APPROVED** [review]

File: tests/test_server.py [hunk 5]
```diff
@@ -316,6 +318,25 @@ def version(self):
...
+    for version in out:
+        vout = str(version).replace(".", "_")
+        if packaging.version.parse(version) > packaging.version.parse("2022.1"):
+            meth = "start_" + vout + "_server"
+            assert (hasattr(server, meth))
+            start = getattr(server, meth)
+            srv = start()
+            assert srv.local_server
+            srv = out[version]()
+            assert srv.local_server
```
Comment (PProfizi) [inline tests/test_server.py:342]: Hey @rlagha,
thanks for the helper, this is great. I guess we should also test that the server version ran is the correct one.
```suggestion
            assert srv.local_server
            assert srv.version == version
            srv = out[version]()
            assert srv.local_server
            assert srv.version == version
```

File: tests/test_server.py [hunk 6]
```diff
@@ -316,6 +318,25 @@ def version(self):
 
     print(MockServer(remote_server).version)
     with pytest.raises(
-        ValueError, match="Error connecting to DPF LegacyGrpcServer with version 1.0"
+            ValueError, match="Could not connect to DPF legacy gRPC server with version 1.0."
```
Comment (PProfizi) [inline tests/test_server.py:321]: @rlagha why did you change the expected error?

Comment (PProfizi) [inline tests/test_server.py:321]: This needs to be reverted as this is only what we expect from the server. Please change it server side if it needs changing.

File: tests/test_server.py (hunk 3 above)
Comment (rlagha) [reply tests/test_server.py:321]: @PipKat , thanks @PipKat , we will fix this in the server side.

**Review by PProfizi — APPROVED** [review]
